addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/ChannelTypeUtils.java

```diff
@@ -86,31 +86,31 @@ public static State toOnOffType(@Nullable Integer value) {
         return value != null ? (value == 1 ? OnOffType.ON : OnOffType.OFF) : UnDefType.NULL;
     }
 
-    public static State toQuantityType(@Nullable Float value, @Nullable Unit<?> unit) {
-        return (value == null || unit == null) ? UnDefType.NULL : toQuantityType(new BigDecimal(value), unit);
+    public static State toQuantityType(@Nullable Float value, Unit<?> unit) {
+        return (value == null) ? UnDefType.NULL : toQuantityType(new BigDecimal(value), unit);
```

Comment:
Maybe remove the parenthesis around the check too as it's only 1 check.

Reply:
I did it.